engine/runtime/src/test/scala/org/enso/compiler/test/semantic/ImportExportTest.scala

```diff
@@ -0,0 +1,566 @@
+package org.enso.compiler.test.semantic
+
+import org.enso.compiler.core.IR
+import org.enso.compiler.data.BindingsMap
+import org.enso.compiler.pass.analyse.BindingAnalysis
+import org.enso.interpreter.runtime
+import org.enso.interpreter.runtime.EnsoContext
+import org.enso.pkg.QualifiedName
+import org.enso.polyglot.{LanguageInfo, MethodNames, RuntimeOptions}
+import org.graalvm.polyglot.{Context, Engine}
+import org.scalatest.BeforeAndAfter
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+
+import java.io.ByteArrayOutputStream
+import java.nio.file.Paths
+
+/** Tests a single package with multiple modules for import/export resolution.
+  * Checks whether the exported symbols and defined entities metadata of the modules
+  * are correct.
+  */
+class ImportExportTest
```

Comment:
I'd prefer the new tests to be written in Java than Scala, but be it...

Reply:
The reason why I chose Scala instead of Java for this particular test is that in this test, we traverse the IR and its metadata and that is way easier to do in Scala - since all the associated definitions are in Scala. I find IR manipulation in Java very clumsy, not even mentioning that IntelliJ is incapable of proper symbol resolution in such cases.

Reply:
A screenshot of IJ of `ErrorCompilerTest.java` that directly manipulates with IR:

![image](https://github.com/enso-org/enso/assets/14013887/abbc2787-56f6-4ff1-b7a8-f43c09e34ef3)

Reply:
I am always pleased seeing other IDEs having problems understanding Java. I like to point out that NetBeans is WYSIWYG - it understands the source code exactly the same way as `javac` does unlike the other IDEs.

Reply:
Regardless of my silly jokes about IDEs, I believe writing such an amount of new Scala code isn't justified.

The test isn't using anything from existing testing infrastructure - e.g. it is not modification of existing Scala code. In such case the general direction is to: move away from Scala.